### PR TITLE
feat: add json to design token output

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #8287

## Summary

updates token output to change js to json.
also fixes files array in package.json to allow SCSS imports using "~@esri/calcite-design-tokens" in calcite-components

### CODE REVIEW

**Review by jcfranco — CHANGES_REQUESTED** [review]
Since this is dropping the `js` output, this should be marked as a breaking change.

File: packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-import { CalciteJs } from "./javascript.js";
+import { CalciteJson } from "./json.js";
 import { CalciteScss } from "./scss.js";
 
 const formatters: Partial<Record<PlatformFormats, string>> = {
   css: CalciteCss,
   scss: CalciteScss,
   sass: CalciteScss,
-  js: CalciteJs,
+  json: CalciteJson,
```
Comment (jcfranco) [inline packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils.ts:11]: Do we need a custom formatter or can we use the predefined formatter for JSON?

Comment (jcfranco) [reply packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils.ts:11]: Did a shallow test and it looks like contents are identical.

**Comment by jcfranco** [comment]
> Since this is dropping the js output, this should be marked as a breaking change.

Alternatively, we could keep the JS output and add the JSON output.

**Comment by jcfranco** [comment]
Sorry for not catching this sooner, but can you also update the spec tests?

**Comment by alisonailea** [comment]
> Sorry for not catching this sooner, but can you also update the spec tests?

opened a different PR to fix the snapshot tests but added the test for the JSON output here

File: packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils.ts (hunk 1 above)
Comment (alisonailea) [reply packages/calcite-design-tokens/support/token-transformer/styleDictionary/formatter/utils.ts]: updated to use "json"

File: packages/calcite-design-tokens/package.json [hunk 2]
```diff
@@ -9,31 +9,10 @@
     "tokens"
   ],
   "type": "module",
-  "main": "./dist/js/global.js",
-  "types": "./dist/js/global.d.ts",
+  "main": "./dist/es6/global.js",
```
Comment (jcfranco) [inline packages/calcite-design-tokens/package.json:12]: Does this need to be reverted since we're keeping the JS output?

**Review by jcfranco — APPROVED** [review]
LGTM! 🚀

Since this is now adding the JSON output, can you update the PR title similar to bf9e03a7b0629d1a3db18c6b104f665666167fe9 before merging?

File: packages/calcite-design-tokens/package.json (hunk 2 above)
Comment (jcfranco) [reply packages/calcite-design-tokens/package.json:12]: Makes sense, but this is now unrelated to this PR and I _think_ falls under breaking change territory.

**Pull request merged**